src/physics.cpp

```diff
@@ -1160,4 +1166,55 @@ void sample_secondary_photons(Particle& p, int i_nuclide)
   }
 }
 
+// split a particle on the basis of the statistical weight
+// of the current location	
+void split_particle(Particle& p)
+{
+  // skip dead or no energy
+  if (p.E() <= 0 || !p.alive()) return;
+  
+  // get the paramters 
+  auto params = settings::ww_settings->get_params(p);
+  double weight = p.wgt();
+
+  // first check to see if particle should be killed for
+  // weight cutoff
+  if ( p.wgt() < params.weight_cutoff ) {
+    p.alive() = false;
+    p.wgt() = 0.0;
+    p.wgt_last() = p.wgt();
+    return;
+  }
+  
+  // if particles weight is above the weight window 
+  // split until they are within the window
+  if (weight > params.upper_weight) {
+    double n_split = weight/params.upper_weight;  
+    n_split = std::min(std::ceil(n_split), double(params.max_split));
+        
+    // possibility of some round off 
+    for (int l = 0; l < int(n_split) - 1; l++)  { 
+      p.create_secondary(weight/n_split, p.u(), p.E(), p.type()); 
+    }
+    // todo maybe weight should be weight - sum of child weight
+    p.wgt() = weight/n_split; 
+    p.wgt_last() = p.wgt();
+   
+    // if the particle weight is below the window  
+    // roulette until the weight is high enough
+  } else if (weight <= params.lower_weight) {  
+    double n_split = std::max(1./params.max_split, weight/params.survival_weight);
+
+    if (prn(p.current_seed()) <= n_split)  {
+      p.wgt() /= n_split;
```

Comment:
I'm thinking we can reduce the number of division operations here:

```suggestion
    double n_split = std::min(params.max_split, params.survival_weight/weight);

    if (n_split*prn(p.current_seed()) > 1)  {
      p.wgt() *= n_split;
```

I think it also is a little more clear that we're increasing the weight in this situation. Does this cause any roundoff issues though?

Reply:
Roundoff is potentially an issue here yes, it will usually be the case in fact that there will be some roundoff, making sure the primary particle accrues the difference is likely the best option.